### PR TITLE
CDAP-18132 add support to add default connections

### PR DESCRIPTION
**Pull request by yaojiefeng**

### CODE REVIEW

File: cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/ServiceSpecificationCodec.java [hunk 1]
```diff
@@ -72,8 +72,9 @@ public ServiceSpecification deserialize(JsonElement json, Type typeOfT,
                                                                     HttpServiceHandlerSpecification.class);
     Resources resources = context.deserialize(jsonObj.get("resources"), Resources.class);
     int instances = jsonObj.get("instances").getAsInt();
+    Map<String, String> properties = deserializeMap(jsonObj.get("properties"), context, String.class);
```
Comment (albertshau) [inline cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/ServiceSpecificationCodec.java:75]: does this return an empty map if properties is null? Need to protect against specs that we written by an older version of CDAP.

File: cdap-app-templates/cdap-etl/cdap-data-pipeline-base/src/main/java/io/cdap/cdap/datapipeline/service/StudioService.java [hunk 2]
```diff
@@ -40,5 +61,51 @@ protected void configure() {
...
+
+      ConnectionId connectionId = new ConnectionId(namespaceSummary, creationRequest.getName());
+      long now = System.currentTimeMillis();
+      Connection connectionInfo = new Connection(
+        creationRequest.getName(), connectionId.getConnectionId(), creationRequest.getPlugin().getName(),
+        creationRequest.getDescription(), true,
+        creationRequest.getName().equals(connectionConfig.getDefaultConnection()) ? true : false,
+        now, now, creationRequest.getPlugin());
+      try {
+        connectionStore.saveConnection(connectionId, connectionInfo, false);
```
Comment (albertshau) [inline cdap-app-templates/cdap-etl/cdap-data-pipeline-base/src/main/java/io/cdap/cdap/datapipeline/service/StudioService.java:105]: this is different behavior than before, where these connections will now be written to the store and persist, even if the default connections change. Is that intended?

Comment (albertshau) [reply cdap-app-templates/cdap-etl/cdap-data-pipeline-base/src/main/java/io/cdap/cdap/datapipeline/service/StudioService.java:105]: I remembered the existing behavior incorrectly, ignore this comment.

**Review by albertshau — APPROVED** [review]

File: cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/ServiceSpecificationCodec.java (hunk 1 above)
Comment (yaojiefeng) [reply cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/ServiceSpecificationCodec.java:75]: Yes, it will return empty map

**Pull request merged**